app/ui.js

```diff
@@ -965,9 +983,171 @@ var UI;
 /* ------^-------
  *     /XVP
  * ==============
- *   CLIPBOARD
+ *    AST2100
  * ------v------*/
 
+        ast2100_handleVideoSettingsChanged: function (settings) {
+            if (!UI._ast2100_videoSettingsInitialized)
+                UI.ast2100_setDefaultSettings(settings);
+            UI.ast2100_updateVideoSettings(settings);
+        },
+        
+        // Called the first time we receive a FramebufferUpdate object.  Responsible
+        // for telling the server about any configured default settings.
+        ast2100_setDefaultSettings: function (settings) {
+            // Convert the settings that noVNC has been configured to set for all
+            // AST2100 servers to the familiar videoSettings format.
+            var defaultQuality = parseInt(UI.ast2100_quality);
+            var defaultSettings = {
+                quantTableSelectorLuma: defaultQuality,
+                quantTableSelectorChroma: defaultQuality,
+                subsamplingMode: parseInt(UI.ast2100_subsamplingMode)
```

Comment:
We can wait with hooking up the Tight stuff (shouldn't be that hard though), but the API and UI should be generic from day one IMO.